### ISSUE TITLE
Fallback function in addConditionWithOptionalFallback is never run

### ISSUE THREAD

**Issue by keimoon** [japgolly/scalajs-react#621]
We've just updated to 1.5.0 and found out the following code does not work as intended:

```
val routes: RouterConfig[Page] = RouterConfigDsl[Page].buildConfig { implicit dsl =>
    import dsl._ // scalastyle:ignore import.grouping underscore.import

    val publicRoutes = (emptyRule
      | staticRoute("#/404", StaticNonAuthPage.NotFound) ~> renderR(_ => <.div("Not found"))
      | staticRoute("#/403", StaticNonAuthPage.NoAccess) ~> renderR(_ => <.div("Login required")))

    val privateRoutes = (emptyRule
      | staticRoute("#/dashboard", StaticAuthPage.Dashboard) ~> renderR(_ => <.div("Dashboard")))
      .addConditionWithOptionalFallback(
        _ => {
          CallbackTo {
            false
          }
        },
        _ => {
          Some(redirectToPage(StaticNonAuthPage.NoAccess)(SetRouteVia.HistoryReplace))
        }
      )

    (trimSlashes | publicRoutes | privateRoutes)
      .notFound { _ =>
        redirectToPage(StaticNonAuthPage.NotFound)(SetRouteVia.HistoryReplace)
      }
  }
```

In previous version, if we go to `#/dashboard`, the fallback in addCondition will be evaluated and redirect to `NoAccess`. However in current version it is redirected to `NotFound`.

**Comment by japgolly** [comment]
Thanks for the bug report @keimoon . We have tests for this behaviour but in a major oversight, the explicit fallback behaviour was the same as the default fallback behaviour. I've fixed this in #624 . I'll merge it and get 1.5.1 released in very soon.

**Comment by japgolly** [comment]
Merged. Subscribe to github releases to get notified when 1.5.1 is released.

Also FYI, you can just use `addConditionWithFallback(Callback(false), redirectToPage(StaticNonAuthPage.NoAccess)(SetRouteVia.HistoryReplace))` now.
